File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTAttributeParameterDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTDataType } from "../../DataDictionary/TTDataType";
import { TTAttributeDef } from "./TTAttributeDef";
import { AttributeParameterSubtypeEnum } from "../../Utils/Enums/AttributeParameterSubtypeEnum";
import { AttributeParameterTargetEnum } from "../../Utils/Enums/AttributeParameterTargetEnum";
import { TTObjectDef } from "../DefinitionManagement/TTObjectDef";

export class TTAttributeParameterDef {
    public AttributeDef: TTAttributeDef;
    public AttributeDefID: Guid;
    public ParameterDefID: Guid;
    public Name: string;
    public Description: string;
    public DataType: TTDataType;
    public DataTypeID: Guid;
    public IsRequired: boolean;
    public ObjectDef: TTObjectDef;
    public ObjectDefID: Guid;
    public SubType: AttributeParameterSubtypeEnum;
    public Target: AttributeParameterTargetEnum;
    public ToString(): string {
        return null;
    }
}

File: AtlasModel/Enterprise/TTAttributeParameterDef.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTAttributeParameterDef
    {
        public Guid ParameterDefId { get; set; }
        public Guid AttributeDefId { get; set; }
        public string Name { get; set; }
        public Guid? DataTypeId { get; set; }
        public Guid? ObjectDefId { get; set; }
        public AttributeParameterTargetEnum Target { get; set; }
        public bool IsRequired { get; set; }
        public short CheckOutStatus { get; set; }
        public string Description { get; set; }
    }
}